python/tests/res/job_queue/test_jobmanager.py

```diff
@@ -476,5 +478,79 @@ def test_job_list_to_log_ordering(self):
             # Verify _ordered_job_map_values is giving us values in same order as job_list
             self.assertEqual(job_manager.job_list, vals)
 
+    def test_exit_signals(self):
+        exit_signals = (0, 1, 2, 42) + tuple(range(128, 140)) + (240,)
+        for exit_signal in exit_signals:
+            area_name = gen_area_name(
+                "exit_signal_{}".format(exit_signal), create_jobs_json
+            )
+            with TestAreaContext(area_name):
+                exit_script_name = "exit.py"
+                with open(exit_script_name, "w") as f:
+                    f.write("\n".join((
+                        "#!/usr/bin/env python",
+                        "import sys",
+                        "sys.exit({})".format(exit_signal),
+                    )))
+                st = os.stat(exit_script_name)
+                os.chmod(exit_script_name, st.st_mode | stat.S_IEXEC)
+                exit_exec = os.path.join(os.getcwd(), exit_script_name)
+
+                joblist = [
+                    {
+                        "name" : "EXITER",
+                        "executable" : exit_exec,
+                        "argList" : [],
+                    }
+                ]
+                create_jobs_json(joblist)
 
+                jobm = JobManager()
+                for job in jobm:
+                    exit_status, msg = jobm.runJob(job)
+                    self.assertEqual(exit_status, exit_signal)
+
+    def test_exit_signals_pid(self):
+        exit_signals = (0, 1, 2, 42) + tuple(range(128, 140)) + (240,)
+        for exit_signal in exit_signals:
+            area_name = gen_area_name(
+                "exit_signal_{}".format(exit_signal), create_jobs_json
+            )
+            with TestAreaContext(area_name):
+                exit_script_name = "exit.py"
+                run_script_name = "run.py"
+                with open(run_script_name, "w") as f:
+                    f.write("\n".join((
+                        '#!/usr/bin/env python',
+                        'import os',
+                        'from res.job_queue import JobManager',
+                        'jobm = JobManager()',
+                        'for job in jobm:',
+                        '    exit_status, msg = jobm.runJob(job)',
+                        'if exit_status != {}:'.format(exit_signal),
```

Comment:
Should this one not be indented one line? It probably does not matter as there is only one job in the tests, so checking after the loop has the same effect. Nevertheless...

Reply:
Indenting the if statement seems reasonable, yes